pkg/executor/remote.go

```diff
@@ -105,15 +118,40 @@ func (ex *Remote) Download(ctx context.Context, remote, local string, opts *UpDo
 		return fmt.Errorf("failed to split hostAddr and port: %w", err)
 	}
 
-	req := sftpReq{
-		client:     ex.client,
-		localFile:  local,
-		remoteFile: remote,
-		mkdir:      opts != nil && opts.Mkdir,
-		remoteHost: host,
-		remotePort: port,
+	var mkdir bool
+	var exclude []string
+
+	if opts != nil {
+		mkdir = opts.Mkdir
+		exclude = opts.Exclude
+	}
+
+	remoteFiles, err := ex.findMatchedFiles(remote, exclude)
+	if err != nil {
+		return fmt.Errorf("failed to list remote files by glob for %s: %w", remote, err)
+	}
+
+	for _, remoteFile := range remoteFiles {
+		localFile := local
+		if filepath.Base(remote) != filepath.Base(remoteFile) {
```

Comment:
I don't understand this if, probably a comment will help

Reply:
Here I compare a glob `match` with the `remote` which can be a filename of a glob. Basically, it makes the same as following code snipped does in `Upload` in `Remote` executor. Still, I cannot rely on `len(remoteFiles)` because `exclude` is already applied to the glob result.

```go
remoteFile := remote
if len(matches) > 1 { // if there are multiple files, treat remote as a directory
	remoteFile = filepath.Join(remote, filepath.Base(match))
}
```
But, definitely, it worth it to add a comment.